bin/reth/src/db/mod.rs

```diff
@@ -241,3 +228,93 @@ impl<'a, DB: Database> DbTool<'a, DB> {
         Ok(())
     }
 }
+
+impl<'a, E: EnvironmentKind> DbTool<'a, Env<E>> {
+    /// Lists the contents of a db table in a human-readable format and print to stdout.
+    fn list_table(&mut self, args: &ListArgs) -> Result<()> {
+        macro_rules! list_tables {
+            ($arg:expr, $start:expr, $len:expr => [$($table:ident),*]) => {
+                match $arg {
+                    $(stringify!($table) => {
+                        let map = self.list::<tables::$table>($start, $len)?;
+                        let mut table = ComfyTable::new();
+                        table.load_preset(comfy_table::presets::ASCII_MARKDOWN);
+                        for (key, value) in map.into_iter() {
+                            let mut row = Row::new();
+                            match serde_json::to_value(&value)? {
+                                serde_json::Value::Object(map) => {
+                                    if table.header().is_none() {
+                                        table.set_header([vec![&"Key".to_owned()], map.keys().collect()].concat());
+                                    }
+
+                                    row.add_cell(Cell::new(format!("{key:?}")));
+                                    map.values().for_each(|v| {
+                                        row.add_cell(Cell::new(v));
+                                    });
```

Comment:
RFC- This is not the cleanest way to do this, but it involves the least amount of changes to the tables' `Value` types (they just must implement serde's `Serialize` trait) from what I've found.

A few caveats here:
* We lose type information upon serialization into JSON. For ex, `H256`s and `U256`s are both serialized as strings. For ex, this means we have to show all 32 bytes of a serialized `U256` value here. 
*  For tables that have a `Value` type with many fields, i.e. `Headers`, the issue of being able to display all of the information within a table in a clean way is also still there (maybe we just display certain fields in these cases?)

Another option I messed around with a few days ago is to use a macro like this to define the structs that will be table values:
```rs
macro_rules! with_field_names {
    (struct $name:ident { $($field_name:ident : $field_type:ty),* }) => {
        struct $name {
            $($field_name : $field_type),*
        }

        impl $name {
            const fn field_names() -> &'static [&'static str] {
                const NAMES: &'static [&'static str] = &[$(stringify!($field_name)),*];
                NAMES
            }
        }
    }
}

with_field_names! {
    struct MyStruct {
        a: u64,
        b: String
        // ...
    }
}
```
But I'm not a huge fan of this. If you guys have a better idea on how to accomplish this, lmk! (cc: @gakonst)